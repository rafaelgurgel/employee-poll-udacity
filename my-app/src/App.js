import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Route, Routes } from "react-router-dom";
import { handleInitialData } from "./actions/shared"; 
import Nav from "./components/nav/Nav";
import Login from "./components/login/Login";
import Dashboard from "./components/dashboard/Dashboard";
import Leaderboard from "./components/leaderboard/LeaderBoard";
import PollDetails from "./components/poll_details/PollDetails";
import NewQuestion from "./components/new_question/NewQuestion";
import NotFound from "./components/not_found/NotFound";
import ProtectedRoute from "./components/protected_route/ProtectedRoute";

export default function App() {
  const dispatch = useDispatch();
  const authedUser = useSelector((state) => state.authedUser);

  useEffect(() => {
    dispatch(handleInitialData());
  }, [dispatch]);

  return (
    <div className="container">
      <Nav />
      <Routes>
        {/* Login route */}
        <Route path="/" element={<Login />} />

        {/* Protected routes */}
        <Route
          path="/home"
          element={
            <ProtectedRoute authedUser={authedUser}>
              <Dashboard />
            </ProtectedRoute>
          }
        />
        <Route
          path="/leaderboard"
          element={
            <ProtectedRoute authedUser={authedUser}>
              <Leaderboard />
            </ProtectedRoute>
          }
        />
        <Route
          path="/add"
          element={
            <ProtectedRoute authedUser={authedUser}>
              <NewQuestion />
            </ProtectedRoute>
          }
        />
        <Route
          path="/questions/:id"
          element={
            <ProtectedRoute authedUser={authedUser}>
              <PollDetails />
            </ProtectedRoute>
          }
        />

        {/* Catch-all routes */}
        <Route path="/notfound" element={<NotFound />} />
        <Route path="*" element={<NotFound />} />
      </Routes>
    </div>
  );
}
