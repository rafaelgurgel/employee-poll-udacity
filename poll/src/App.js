import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Route, Routes } from 'react-router-dom'
import { handleInitialData } from './actions/shared'
import Nav from './components/Nav'
import Login from './components/Login'
import Dashboard from './components/Dashboard'
import Leaderboard from './components/LeaderBoard'
import PollDetails from './components/PollDetails'
import NewQuestion from './components/NewQuestion'
import NotFound from './components/NotFound'
import ProtectedRoute from './components/ProtectedRoute'

export default function App() {
  const dispatch = useDispatch()
  const authedUser = useSelector((state) => state.authedUser)

  useEffect(() => {
    dispatch(handleInitialData())
  }, [dispatch])

  return (
    <div className="container">
      <Nav />
      <Routes>
        <Route path="/" element={<Login />} />
        <Route
          path="/home"
          element={
            <ProtectedRoute authedUser={authedUser}>
              <Dashboard />
            </ProtectedRoute>
          }
        />
        <Route
          path="/leaderboard"
          element={
            <ProtectedRoute authedUser={authedUser}>
              <Leaderboard />
            </ProtectedRoute>
          }
        />
        <Route
          path="/add"
          element={
            <ProtectedRoute authedUser={authedUser}>
              <NewQuestion />
            </ProtectedRoute>
          }
        />
        <Route
          path="/questions/:id"
          element={
            <ProtectedRoute authedUser={authedUser}>
              <PollDetails />
            </ProtectedRoute>
          }
        />
        <Route path="*" element={<NotFound />} />
      </Routes>
    </div>
  )
}
