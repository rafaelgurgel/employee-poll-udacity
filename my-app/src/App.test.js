import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { configureStore } from '@reduxjs/toolkit'
import { MemoryRouter } from 'react-router-dom'
import App from './App'
import '@testing-library/jest-dom'

// Mock reducer
const mockReducer = (state = {
  authedUser: 'sarahedo',
  users: {
    sarahedo: { id: 'sarahedo', name: 'Sarah Edo', answers: { q1: 'optionOne' }, avatarURL: null },
  },
  questions: {
    q1: {
      id: 'q1',
      author: 'sarahedo',
      optionOne: { votes: ['sarahedo'], text: 'Option 1' },
      optionTwo: { votes: [], text: 'Option 2' },
    },
  },
}) => state

const mockStore = configureStore({
  reducer: mockReducer,
})

describe('App Component', () => {
  it('renders the PollDetails page for authenticated users', () => {
    const { getByText } = render(
      <Provider store={mockStore}>
        <MemoryRouter initialEntries={['/questions/q1']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(getByText(/option 1/i)).toBeInTheDocument() // Assuming this text is in PollDetails
  })
})
