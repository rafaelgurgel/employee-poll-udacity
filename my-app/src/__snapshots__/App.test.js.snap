// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <nav
      class="nav"
    >
      <a
        href="/home"
      >
        Home
      </a>
      <a
        href="/add"
      >
        New Poll
      </a>
      <a
        href="/leaderboard"
      >
        Leaderboard
      </a>
      <div
        class="nav-profile"
      >
        <img
          alt="Avatar of Sarah Edo"
          class="nav-avatar"
          src="defaultAvatar.svg"
        />
        <span>
          Hello, Sarah Edo
        </span>
        <button
          class="logout-button"
        >
          Logout
        </button>
      </div>
    </nav>
    <div
      class="polldetails-container"
    >
      <h3
        class="polldetails-title"
      >
        Poll by Sarah Edo
      </h3>
      <img
        alt="Avatar of Sarah Edo"
        class="polldetails-avatar"
        src="defaultAvatar.svg"
      />
      <h4
        class="polldetails-subtitle"
      >
        Would You Rather...
      </h4>
      <div>
        <div
          class="polldetails-results"
          style="font-weight: bold; color: green;"
        >
          Option 1
          <br />
          1 vote(s) (100.00%)
        </div>
        <div
          class="polldetails-results"
        >
          Option 2
          <br />
          0 vote(s) (0.00%)
        </div>
      </div>
      <br />
      <button
        class="polldetails-back"
      >
        Back
      </button>
    </div>
  </div>
</DocumentFragment>
`;
